include/uv/win.h

```diff
@@ -287,11 +287,15 @@ typedef struct {
   DWORD tls_index;
 } uv_key_t;
 
-#define UV_ONCE_INIT { 0, NULL }
+#define UV_ONCE_INIT { .init_once = INIT_ONCE_STATIC_INIT }
```

Comment:
This can't land in the v1.x branch unless they're exactly equivalent, and I don't think they are. INIT_ONCE is pointer-sized, `ran` is byte-sized. Existing downstream code won't fully initialize the field.

Or rather, it _may_, depending on what machine code the compiler generated (it's unspecified if or how holes in structs are initialized), but you can't rely on that.

Reply:
Hum, I thought the union + struct do the trick because HANDLE is also a pointer?

Or you mean the initialization itself?

Reply:
The initialization itself. If I write code against libuv 1.48.0 and then upgrade to 1.49.0 that contains this change, my existing compiled code should keep working. But it may or may not depending on whether the compiler translated it to something like this:
```c
memset(&once, 0, sizeof(once)); // no gap
```
Vs.
```c
once.ran = 0;
// uninitialized gap
once.event = 0;
```

Reply:
Gotcha. Not sure how to do that TBH. I'll check the windows headers, maybe `{0}` can work, if zeroing out is all they do...

Reply:
Now that I think about it... HANDLE and INIT_ONCE are the same size. It should be safe to replace the former with the latter, as long as you leave the field order unchanged.

Reply:
Ah good point!